src/plugin.js

```diff
@@ -111,32 +122,49 @@ const defaultGetIconSize = (fileName) => {
  * @returns the mime type of the image, as inferred by the file extension.
  */
 const defaultGetIconType = (fileName) => {
-  const extension = fileName.match(/manifest\/icon_(\d+)-\w*\.(png|jpeg|jpg)$/)[2];
+  const match = fileName.match(/manifest\/icon_(\d+)-\w*\.(png|jpeg|jpg)$/);
+  const extension = match && match[2];
+  /* istanbul ignore if */
+  if (!extension) {
+    throw new Error(
+      `Invalid icon extension found ${JSON.stringify(extension)} in filename ${JSON.stringify(
+        fileName,
+      )}`,
+    );
+  }
   return `image/${extension}`;
 };
 
+/**
+ * @typedef Config
+ * @property {ManifestContent} content represents an object that will be validated and converted to JSON as the contents of the manifest file.
+ * @property {string} destination is an output path where the manifest file should be written.
+ * @property {(filename: string) => boolean} [isAssetManifestIcon] a function to determine if a webpack asset should be included as an icon in the web app manifest. The function accepts a `filename` parameter and returns true or false.
+ * @property {(filename: string) => Dimensions} [getIconSize] a function to determine the icon size of any asset that passes the check `isAssetManifestIcon()`. The function accepts a `fileName` parameter and returns an object `{ width, height }`.
+ * @property {(filename: string) => string} [getIconType] a function to determine the type of any asset that passes the check `isAssetManifestIcon()`. The function accepts a `fileName` parameter and returns a string describing the mime type of the asset, ex. "image/png".
+ *
+ * @typedef Dimensions
+ * @property {number} width
+ * @property {number} height
+ *
+ * @typedef ManifestContent
+ * @property {unknown} [name]
+ * @property {unknown} [short_name]
+ * @property {unknown} [start_url]
+ * @property {unknown} [display]
+ * @property {unknown} [background_color]
+ * @property {unknown} [theme_color]
+ * @property {unknown} [description]
+ * @property {unknown} [icons]
+ * @property {unknown} [related_applications]
+ * @property {unknown} [prefer_related_applications]
```

Comment:
Tempting to parse to this format:

https://developer.mozilla.org/en-US/docs/Web/Manifest

Reply:
https://www.npmjs.com/package/@types/web-app-manifest